### PR TITLE
Create Datagen_Project-milestone_1.md

### PR DESCRIPTION
**Pull request by Lord-Nymphis**
# Milestone Delivery Checklist

- [x] The [milestone-delivery-template.md](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/deliveries/milestone-delivery-template.md) has been copied and updated.
- [x] The [invoice form :pencil:](https://forms.gle/LSRr7PCjBpEbKGh89) has been filled out for this milestone.
- [x] This pull request is being made by the same account as the accepted application.
- [x] In case of acceptance, the payment will be transferred to the BTC/ETH payment address provided in the application.
- [x] The delivery is according to the [Guidelines for Milestone Deliverables](https://github.com/w3f/Grants-Program/blob/master/docs/milestone-deliverables-guidelines.md).

Link to the application pull request: https://github.com/w3f/Grants-Program/pull/980

### CODE REVIEW

File: deliveries/Datagen_Project-milestone_1.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Maintenance Delivery :mailbox:
+
+**The [invoice form :pencil:](https://docs.google.com/forms/d/e/1FAIpQLSfmNYaoCgrxyhzgoKQ0ynQvnNRoTmgApz9NrMp-hd8mhIiO0A/viewform) has been filled out correctly for this delivery**  
+
+* **Application Document:** https://github.com/w3f/Grants-Program/blob/master/applications/Datagen_Project.md
+* **Delivery Number:** 1
```
Comment (alxs) [inline deliveries/Datagen_Project-milestone_1.md:6]: This will fix the checks/integration

```suggestion
* **Milestone Number:** 1
```

**Comment by alxs** [comment]
Thank you for your patience @Lord-Nymphis. One of us will take a look at your application as soon as possible.

**Comment by dsm-w3f** [comment]
@Lord-Nymphis thank you for the milestone submission and my apologies for the late response. Please see [the evaluation document](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/datagen_1_dsm-w3f.md) and provide proper answers and fixes. After that, let me know to continue with the evaluation.

**Comment by Lord-Nymphis** [comment]
Thank you @dsm-w3f , we'll try to fix the problems you underlined by the end of the week and also to properly address the specific doubts

**Comment by viac92** [comment]
Hi @dsm-w3f thank you for your evaluation! 
I'm working to fix the problems, I will let you kwon when I'm done :)

**Comment by viac92** [comment]
Hi @dsm-w3f I did some update to match your requests:

### Testing guide

- I don't understand if you mean when you use polkadot.js app or when you test with `cargo test`, if you use `cargo test` the blockchain is spin up with the `run_to_block ` function, with the polkadot.js please follow the instructions in the readme, I made a quick [video demo](https://www.youtube.com/watch?v=4gwC2lOTazY) to make the process more clear. 
- Yes I know there are fixed behavior in the code, now it's just for helping the testing process I will update and made flexible in the second milestone where I will setup the connections with the two blockchains.

### Docker 
I remove the docker-compose and setup the Dockerfile, follow the testing guide.

### Test coverage 
I add more tests for the `random_node_selector` pallet.

### Static analysis 
This warnings are from the node-template, many are from rpc, I will fix theme in the next milestone where I will implement custom rpc.

### Code Documentation 
I fix the error.

**Comment by Lord-Nymphis** [comment]
Hi @alxs , @dsm-w3f  do you have already some feedbacks on our updates :) ?

**Comment by dsm-w3f** [comment]
@Lord-Nymphis and @viac92 my apologies for the late response. You are the next evaluation in my backlog. I'll probably provide a feedback about the changes today or tomorrow.

**Comment by dsm-w3f** [comment]
@Lord-Nymphis and @viac92 thank you for the improvements and fixes. The milestone was approved. Great work! Please notice in [the evaluation document](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/datagen_1_dsm-w3f.md) a small improvement in docker that would be nice to have in the next milestone delivery. I'll forward internally your invoice and the payment should take place within two weeks (could happen before).

**Comment by fededubbi** [comment]
Hi @Lord-Nymphis,
I noticed that on your invoice is not included the payment address.
Could you please add it and resubmit the invoice?

Many thanks,
Federica

**Comment by Lord-Nymphis** [comment]
Hi @fededubbi ,
nice to e-meet you!

I think it was only in the application and not in the invoice,
Just added it.

Have a good day!

**Comment by RouvenP** [comment]
hi @Lord-Nymphis we transferred the payment today

**Comment by Lord-Nymphis** [comment]
Hi @RouvenP , thank you. Seen.

**Pull request merged**